slides/04_quality.qmd

```diff
@@ -1,10 +1,655 @@
+    check](https://r-pkgs.org/R-CMD-check.html) from R and capture results
+-   [devtools](https://cran.r-project.org/package=devtools): Tools to make
+    developing R packages easier, e.g.,
+    [check()](https://rdrr.io/cran/devtools/man/check.html) automatically builds
+    and checks a source package, using all known best practices
+
+## How to improve the test coverage?
+
+[covr](https://cran.r-project.org/package=covr): Track and report code coverage
+for your package
+
+```{r, echo = TRUE, eval = FALSE}
+library(covr)
+x <- package_coverage()
+report(x)
+```
+
+. . .
+
+![](resources/covr_full.png)
+
+## Example on test coverage
+
+Let's assume we have added a generic function to cat a simulation result:
+
+```{r, echo = TRUE, eval = FALSE}
+#' 
+#' @export
+#'
+cat.SimulationResult <- function(... , file = "", sep = " ", 
+        fill = FALSE, labels = NULL, append = FALSE) {
+    args <- list(...)
+}
+```
+
+. . .
+
+But we forgot to finalize the implementation and didn't create unit tests.
+
+Re-execution of
+
+```{r, echo = TRUE, eval = FALSE}
+#| code-line-numbers: false
+report(package_coverage())
+```
+
+then results in...
+
+## Example on test coverage
+
+![](resources/covr_example_1.png)
+
+## Example on test coverage
+
+We can go into the details by clicking on a file name:
+
+![](resources/covr_example_2.png)
+
+# Code Style
+
+::: clean-code-rule
+CCR#2: Formatting
+:::
+
+![[Photo CC0 by Pavel Danilyuk on
+pexels.com]{.copyright}](thumbnails/styling.jpg){fig-align="center"}
+
+## Why is Code Style important?
+
+::: clean-code-rule
+CCR#2: Formatting
+:::
+
+-   Make the code more readable, maintainable, and consistent
+-   Make it easier for others to understand and contribute to the codebase
+-   Adhering to a consistent code style can reduce the number of errors and make
+    debugging simpler
+-   "Good coding style is like correct punctuation: you can manage without it,
+    butitsuremakesthingseasiertoread." ([The tidyverse style
+    guide](https://style.tidyverse.org))
+
+## How to optimize the code styling?
+
+Two popular R packages support the [tidyverse style
+guide](https://style.tidyverse.org):
+
+-   [styler](https://styler.r-lib.org): interactively restyle selected text,
+    files, or entire projects:
+    -   [style_text](https://styler.r-lib.org/reference/style_text.html)
+    -   [style_file](https://styler.r-lib.org/reference/style_file.html)
+    -   [style_pkg](https://styler.r-lib.org/reference/style_pkg.html)
+-   [lintr](https://github.com/jimhester/lintr): perform automated checks to
+    confirm that you conform to the style guide
+
+The [devtools](https://devtools.r-lib.org) function
+[spell_check](https://devtools.r-lib.org/reference/spell_check.html) runs a
+spell check on text fields in the package description file, manual pages, and
+optionally vignettes.
+
+## Code Styler in RStudio
+
+::: panel-tabset
+## Browse addins
```

Comment:
For me this was not needed, I always just installed the `styler` package in R with `install.packages()` and the addin appeared - maybe easier?

Reply:
The basic idea of these slides were to show how to link the already installed styler with a keyboard shortcut. I optimized the slides a little bit.